Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -166,6 +171,24 @@ public class ConnectionPool {
         unlockPoolLock()
     }
     
+    private func isConnectionPoolPopulated() -> Bool {
```

Comment:
This function conflates two things: "is the connection pool populated?", a boolean test of the current pool status, and "attempt to expand the pool if it's empty".

It's not clear from the function name that it can potentially modify state in this way.

I think logic should be reworked.

Reply:
I think the objective of the function is okay (to check to see if the pool is at zero capacity, and attempt to recover from this if necessary) but the naming needs improvement.  Something like `tryToEnsureCapacityIsNonZero()`.

The problem is that capacity being zero indicates some problem has occurred (network down, DB down) - that all connections have been purged from the pool, as the generator was unable to replace them - and so we can reasonably expect the generator to fail here too.  At which point, we need to exit `take()` early so as not to block.

However, once the DB is reachable again, this function can kick-start the process of growing the pool back up (only possible once there is capacity > 0, as the 'grow the pool' code is within the `semaphore.wait()`). 

One other possibility that I discussed with @ddunn2 is to move the 'grow the pool' logic to the top of the `take()` function, and using it to also seed the pool with the first item if it has become dead.